File: src/AdventureWorks.Domain/Models/StateProvinceModel.cs
using AdventureWorks.Domain.Models.Slim;

namespace AdventureWorks.Domain.Models;

public sealed class StateProvinceModel
{
    public int Id { get; set; }

    public string Code { get; set; }

    public string Name { get; set; }

    public bool IsStateProvinceCodeUnavailable { get; set; }
    
    public CountryRegionModel CountryRegion { get; set; }

    public GenericSlimModel Territory { get; set; }
}

File: apps/angular-web/libs/shared/data-access/src/lib/models/lookup/state-province.model.ts
import { SlimReference } from '../slim-reference.model';
import { CountryRegion } from './country-region.model';

export interface StateProvince {
  readonly id: number;
  readonly code: string;
  readonly name: string;
  readonly isStateProvinceCodeUnavailable: boolean;
  readonly countryRegion: CountryRegion;
  readonly territory: SlimReference;
}
